synapse/handlers/federation_event.py

```diff
@@ -1250,9 +1250,18 @@ async def _check_event_auth(
         # This method should only be used for non-outliers
         assert not event.internal_metadata.outlier
 
+        # first of all, check that the event itself is valid.
         room_version = await self._store.get_room_version_id(event.room_id)
         room_version_obj = KNOWN_ROOM_VERSIONS[room_version]
 
+        try:
+            validate_event_for_room_version(room_version_obj, event)
+        except AuthError as e:
+            logger.warning("While validating received event %r: %s", event, e)
+            # TODO: use a different rejected reason here?
+            context.rejected = RejectedReason.AUTH_ERROR
+            return context
```

Comment:
Looks like moving this up means that we won't call `cache_joined_hosts_for_event` in some situations, but I see #10986 moves it out of this method anyway.

Reply:
yeah. We shouldn't be calling `cache_joined_hosts_for_event` on rejected events anyway, since we won't be sending them out.